Correct and update mapping of D_AGC values

markjfine opened the pull request:
Ref issue[#1775](https://github.com/Hamlib/Hamlib/issues/1775)
 Part 1 : Corrects the mapping of D_AGC values to currently enumerated AGC set, including those likely not available when originally coded.
 
 Part 2 will separately provide alternate mapping, if required, in backends (such as IC746 Pro) that require it.

N0NB commented:
Hopefully we can get some Icom owners to test these changes before I merge them into master.

N0NB commented:
So far we've not seen anything about anyone testing this PR.  It's a bit clumsy to do so, I admit.  I guess it can be merged and then it *might* get tested.

markjfine commented:
I'm in no rush. I have my local git version here so all's good.

Since Github's become more popular, I'm seeing fewer people actually test things in waiting. What they're more prone to do is wait until it hits master then b&m if it doesn't work. They can't be bothered. That's not the way software development is supposed to work, but that's where we are. People want this site to be professionally responsive, but not needing their participation, which is an incorrect assumption.

Give you an example: People have been complaining about automating remote network builds for Airspy for ages. I had some down time, so tried to help Touil out and wrote it for him. The PR's been sitting for weeks since none of the people that actually requested the enhancement have bothered to actually try it.

markjfine commented:
Crap... meant to clean up the unnecessary comments...

N0NB commented:
I don't see unnecessary comments.  Their presence is harmless unless they're wrong.

markjfine commented:
Well, also went back and corrected something I incorrectly assumed about CW Pitch. Apparently only the R75 was using the set mode command. So just the interpolation calculation was improperly applied. I reverted the code and added a special case just for R75.

Again, seems like not the right way to go about this to me. It should be done in the R75 backend file, not in a generic icom.c for one case, but that's what's there. Simple fix.

BTW: This is a good example why I tend to comment things out while testing instead of overwriting. Makes reverting code so much easier.

markjfine commented:
> I don't see unnecessary comments. Their presence is harmless unless they're wrong.

Really a style thing. Makes things cleaner looking and easier to read without having the extra clutter.

GeoBaltz commented:
FYI - ran the cppcheck.sh on merged code, and here are the gripes for icom_alt_agc.c

gwb@stitch:~/OSS/src/Hamlib> grep icom_a cppcheck.log 
rigs/icom/icom_alt_agc.c:417,style,knownConditionTrueFalse,Condition 'lvl_sc==-1' is always false
rigs/icom/icom_alt_agc.c:406,error,uninitvar,Uninitialized variable: cmdbuf
rigs/icom/icom_alt_agc.c:237,style,unusedFunction,The function 'icom_rig_set_level' is never used.
rigs/icom/icom_alt_agc.c:364,style,unusedFunction,The function 'icom_rig_get_level' is never used.

I think the 'error' is sort of a false positive, since cmd_len == 0, so no action needed unless you want to.
73 n3gb
